website/source/docs/plugins/provider.html.md

```diff
@@ -142,7 +142,11 @@ The CRUD operations in more detail, along with their contracts:
 
   * `Create` - This is called to create a new instance of the resource.
       Terraform guarantees that an existing ID is not set on the resource
-      data. That is, you're working with a new resource.
+      data. That is, you're working with a new resource. Therefore, you are
+      responsible for calling `setId` on your `schema.ResourceData` using a
+      value suitable for your resource. This ensures whatever resource
+      state you set on `schema.ResourceData` will be persisted in local state.
+      If you neglect to `setId`, no resource state will be persisted.
```

Comment:
I wonder if it's worth noting explicitly (here or somewhere else) that Terraform considers an empty id to mean "no resource exists", and that any of these methods can actually set it to the empty string to tell Terraform that the resource is gone.